src/orders/views/OrderDetails/OrderNormalDetails/index.tsx

```diff
@@ -415,6 +424,22 @@ export const OrderNormalDetails: React.FC<OrderNormalDetailsProps> = ({
         onClose={closeModal}
         onSend={() => orderInvoiceSend.mutate({ id: params.id })}
       />
+      <OrderManualTransactionDialog
+        open={params.action === "add-manual-transaction"}
+        onClose={closeModal}
+        confirmButtonState={orderAddManualTransaction.opts.status}
+        currency={data?.order?.totalBalance?.currency}
+        onCreateTransaction={({ amount, description }) =>
+          orderAddManualTransaction.mutate({
+            currency: data?.order?.totalBalance?.currency,
+            orderId: id,
+            amount,
+            // hack for GraphQL different types
+            amount2: amount,
+            description,
+          })
+        }
+      />
```

Comment:
you actually copy-paste the same logic in two places, especially the mutate call, can't you use it inside OrderManualTransactionDialog ? You can move it to the dedicated hooks (each one responsibility), and ofc test them separatelly

Reply:
Try to thing how you can avoid copying-pasting the same jsx in two places, I know it may require wider refactoring, but perhaps there is a way to do it. eg in some parent component?

Reply:
I've refactored the duplicated parts in `OrderSendRefund` and in the modal - I didn't do that for the `OrderNormalDetails` and `OrderUnconfirmedDetails` since that would be too much work and could possibly break stuff without knowing it 😅